src/libraries/Microsoft.Extensions.Options/gen/Parser.cs

```diff
@@ -448,6 +448,20 @@ private List<ValidatedMember> GetMembersToValidate(ITypeSymbol modelType, bool s
                 }
             }
 
+            bool validationAttributeIsApplied = validationAttrs.Count > 0 || transValidatorTypeName is not null || enumerationValidatorTypeName is not null;
+
+            if (member.IsStatic)
+            {
+                // generate an error if the member is static and has a validation attribute applied
```

Comment:
> // generate an error if the member is static and has a validation attribute applied [](http://example.com/codeflow?start=15&length=84)

Is it the correct thing to generate error and not just ignoring the statics?

Reply:
Would be good if we check what is the runtime behavior when having the validation attribute on the statics.

Reply:
```
using a validation framework like ASP.NET MVC or ASP.NET Core, the RequiredAttribute alone would not automatically trigger validation for static properties. These frameworks primarily focus on validating instance properties of classes that are involved in HTTP request processing.
```

Considering that, I would say making it as `warning` will make sense. 

@jeffhandley @geeknoid what do you think about that?

Reply:
Turns out it was a warning already (bad copy-paste). Added an additional assertion in tests to ensure that